backend/prompt_studio/prompt_studio_index_manager/models.py

```diff
@@ -78,3 +88,28 @@ class Meta:
                 name="unique_document_manager_profile_manager",
             ),
         ]
+
+
+# Function will be executed every time an instance of IndexManager is deleted.
+@receiver(post_delete, sender=IndexManager)
+def perform_vector_db_cleanup(sender, instance, **kwargs):
+    """Signal to perform vector db cleanup."""
+    logger.info("Performing vector db cleanup")
+    try:
+        # Get the index_ids_history to clean up from the vector db
+        index_ids_history = json.loads(instance.index_ids_history)
+        vector_db_instance_id = str(instance.profile_manager.vector_store.id)
+        # Generate a run_id
+        org_schema = connection.tenant.schema_name
+        util = PromptIdeBaseTool(log_level=LogLevel.INFO, org_id=org_schema)
+        vector_db = VectorDB(
+            tool=util,
+            adapter_instance_id=vector_db_instance_id,
+        )
+        for index_id in index_ids_history:
```

Comment:
NIT: It might be a good idea to catch and ignore exceptions within this loop so that it can proceed for other index_ids

Reply:
Already ran through the scenario, even if the id call fails on vector end, it returns 200

Reply:
@athul-rs that's okay, I meant that let's say we're looping to delete 10 index records and we fail for some reason at the 3rd one, with the current code we won't attempt for the remaining 7 right

Reply:
@chandrasekharan-zipstack Understand your concern, though ideally this shouldn't happen I have now added a catch for these errors to avoid breakage in loop.